pulser/simulation/simresults.py

```diff
@@ -14,63 +14,336 @@
+
+        Args:
+            op (qutip.Qobj): Operator whose expectation value is wanted.
+            fmt (str): Curve plot format.
+            label (str): Axis label.
+        """
+        plt.plot(self._sim_times, self.expect([op])[0], fmt, label=label)
+        plt.xlabel('Time (µs)')
+        plt.ylabel('Expectation value')
+
+    def _get_index_from_time(self, t_float: float, tol: float = 1.e-3) -> int:
+        """Returns closest index corresponding to time t_float.
+
+        Args:
+            t_float (float): Time the time index of which is needed.
+            tol (float): Tolerance for the difference between t_float and
+                closest time.
+        """
+        try:
+            return int(np.where(abs(t_float - self._sim_times) < tol)[0][0])
+        except IndexError:
+            raise IndexError(
+                f"Given time {t_float} is absent from Simulation times within"
+                + f" tolerance {tol}.")
+
+
+class NoisyResults(SimulationResults):
+    """Results of a noisy simulation run of a pulse sequence.
+
+    Contrary to a CleanResults object, this object contains a list of Counter
+    describing the state distribution at the time it was created by using
+    Simulation.run() with a noisy simulation.
+    Contains methods for studying the populations and extracting useful
+    information from them.
+    """
+
+    def __init__(self, run_output: list[Counter],
+                 size: int, basis_name: str,
+                 sim_times: np.ndarray, n_measures: int, dim: int = 2) -> None:
+        """Initializes a new NoisyResults instance.
+
+        Warning:
+            Can't have single-atom Hilbert spaces with dimension bigger
+            than 2 for NoisyResults objects.
+            This is not the case for a CleanResults object, containing states
+            in Hilbert space, but NoisyResults contains a probability
+            distribution of bitstrings, not atomic states
+
+        Args:
+            run_output (list[Counter]): Each Counter contains the
+                probability distribution of a multi-qubits state,
+                represented as a bitstring. There is one Counter for each time
+                the simulation was asked to return a result.
+            size (int): The number of atoms in the register.
+            basis_name (str): Basis indicating the addressed atoms after
+                the pulse sequence ('ground-rydberg' or 'digital' - 'all' basis
+                makes no sense after projection on bitstrings).
+            sim_times (list): Times at which Simulation object returned the
+                results.
+            meas_basis (Optional[str]): The basis in which a sampling
+                measurement is desired.
+            n_measures (int): Number of measurements needed to compute this
+                result when doing the simulation.
+            dim (int): Equal to 2 here, since projections already happened.
```

Comment:
You can remove this one from the signature and the docstring and fix it to `2` on the code.

In fact, I think it's dumb we are even asking for it, considering we can deduce it from the `basis_name`. What if we don't ask for it anywhere and just define it inside `SimulationResults.__init__()` as:
```
self._dim = 3 if basis_name == "all" else 2
```